otter/test/models/test_intents.py

```diff
@@ -25,15 +27,26 @@ def setUp(self):
         """
         Sample group, collection and dispatcher
         """
-        self.log = mock_log()
+        self.log = mock_log().bind(base_log=True)
         self.group = mock_group(None)
-        self.store = iMock(IScalingGroupCollection)
-        self.dispatcher = get_model_dispatcher(self.log, self.store)
 
-        def get_scaling_group(log, tenant_id, group_id):
-            return self.data[(log, tenant_id, group_id)]
+    def get_dispatcher(self, store):
+        return get_model_dispatcher(self.log, store)
 
-        self.store.get_scaling_group.side_effect = get_scaling_group
+    def get_store(self):
+        return iMock(IScalingGroupCollection)
+
+    def perform_with_group(self, eff, expected_lookup, group, disp=None):
+        """Run an effect that will look up group info."""
+        def gsg(log, tenant_id, group_id):
+            assert (log, tenant_id, group_id) == expected_lookup
+            return group
+        store = self.get_store()
+        store.get_scaling_group.side_effect = gsg
+        dispatcher = self.get_dispatcher(store)
+        if disp is not None:
+            dispatcher = ComposedDispatcher([dispatcher, disp])
```

Comment:
Non-blocking nitpick: `[dispatcher, disp]` seems a little confusing. I wonder if any of the two can have better names?